### ISSUE TITLE
Consider removing "async" version of file operations

### ISSUE THREAD

**Issue by djspiewak** [functional-streams-for-scala/fs2#1230]
Pro tip: they aren't actually async! At least, not async in the way we normally think of it. The JVM is faking it, mostly because it isn't actually supported by any underlying platform (some background here: https://www.remlab.net/op/nonblock.shtml). `AsynchronousFileChannel` is backed by an implementation which simply submits tasks to a hidden thread pool; it isn't actually using non-blocking IO in the way that `AsynchronousSocketChannel` does.

In other words, the *non*-async versions of the operations are actually much more controlled (since they use `ContextShift`) and should provide roughly identical performance and throughput.

**Comment by mpilquist** [comment]
👍 from me. I'm surprised that `AsynchronousFileChannel` just dispatches blocking reads to an executor.

**Comment by pchlupacek** [comment]
hm, this is really surprising. Yeah we shall dropping this and make a comment why we need special ContextShift to handle fileIO. 

I am quite confused however why this statement

>  If the system needs time to perform the I/O operation, it will put the task in non-interruptible sleep from the read or write system call.

doesn't imply there is a room for non-blocking tasks, that simply are woken up when disk is ready, instead blocking // sleeping the thread.

**Comment by mpilquist** [comment]
I saw the response on Twitter linking to the Windows API. I'm not convinced that's a strong enough reason to keep the async implementation in fs2 though. I'm afraid folks will err on side of using async variants b/c it sounds better.

**Comment by pchlupacek** [comment]
we could always `correct` async behaviour. That means we could keep using async versions and anyl shift/from/back to main threadpool.

**Comment by mpilquist** [comment]
@pchlupacek Right, but that's what the sync versions do now

**Comment by pchlupacek** [comment]
@mpilquist yeah, the difference is if the platform is able to use the async variant and provide better behaviour, we can still take benefits of it. Not sure if the sync version on platform that is not async deliver better performance.

**Comment by SystemFw** [comment]
I agree with @mpilquist , at this point we should probably just remove it

**Comment by pchlupacek** [comment]
cool. lets do it

**Comment by djspiewak** [comment]
For posterity, I have confirmed that on Windows Vista (and Server 2008) and above, `AsynchronousFileChannel` does in fact use truly async operations. Not that I disagree with the conclusions here, just wanted to make sure the information was out there.
